src/Microsoft.Net.Http.Headers/NameValueHeaderValue.cs

```diff
@@ -142,6 +147,20 @@ public override bool Equals(object obj)
             }
         }
 
+        public StringSegment GetDecodedValue()
+        {
+            return HeaderUtilities.RemoveQuotes(_value);
+        }
+
+        public void SetAndEncodeValue(StringSegment value)
+        {
+            if (_quotedValue)
+            {
+                value = $"\"{value}\"";
```

Comment:
1. See above comment.
2. I am not sure what other utilities are necessary besides unquoting. I will fix this to handle the double quote scenario.